scripts/shardChecker.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+
+	mvxcore "github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/data/alteredAccount"
+	"github.com/ElrondNetwork/elrond-go-core/data/firehose"
+	"github.com/tidwall/gjson"
+)
+
+func checkShardBlock(hyperBlockNonce uint64, address string, txHash string) error {
+	log.Info("checking shard block...")
+
+	resp, err := http.Get(fmt.Sprintf("%s/hyperblock/by-nonce/%d", proxyUrl, hyperBlockNonce))
+	if err != nil {
+		return err
+	}
+
+	log.Info("checking data from hyperblock", "hyperBlockNonce", hyperBlockNonce)
+	defer func() {
+		err = resp.Body.Close()
+		if err != nil {
+			log.Warn("could not close response body", "error", err)
+		}
+	}()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	shardBlocks := gjson.Get(string(body), "data.hyperblock.shardBlocks").Array()
+	if len(shardBlocks) != 1 {
+		return fmt.Errorf("checkShardHeader: should only have one shard, but got %d", len(shardBlocks))
```

Comment:
should only have one shard or block?

Reply:
One shard only. It is described in the app usage and script usage as well:

script:
`This script will start a firehose node in the provided shard(metachain/shard).
  The only acceptable parameters are shard(will use shard 0) and metachain for the current test configuration`

checker:
`app.Usage = "This tool only works if a local testnet and a firehose node are started. See firehose-node.sh and local-testnet.sh scripts"`

Will also add a Readme here to make things clearer